test/testcontext/testcontext.go

```diff
@@ -0,0 +1,807 @@
+			require.True(ctx.T, ok, "you must specify a work creator ID for the \"%s\" field in %+v", workitem.SystemCreator, ctx.WorkItems[i].Fields)
+		}
+		creatorIDStr, ok := ctx.WorkItems[i].Fields[workitem.SystemCreator].(string)
+		require.True(ctx.T, ok, "failed to convert \"%s\" field to string", workitem.SystemCreator)
+		creatorID, err := uuid.FromString(creatorIDStr)
+		require.Nil(ctx.T, err, "failed to convert \"%s\" field to uuid.UUID", workitem.SystemCreator)
+
+		err = workitem.NewWorkItemRepository(ctx.db).CreateFromModel(ctx.ctx, ctx.WorkItems[i], creatorID)
+		require.Nil(ctx.T, err, "failed to create work item: %+v", ctx.WorkItems[i])
+	}
+}
+
+func makeWorkItemLinks(ctx *TestContext) {
+	if ctx.info[kindWorkItemLinks] == nil {
+		return
+	}
+	ctx.WorkItemLinks = make([]*link.WorkItemLink, ctx.info[kindWorkItemLinks].numInstances)
+	for i := range ctx.WorkItemLinks {
+		id := uuid.NewV4()
+		ctx.WorkItemLinks[i] = &link.WorkItemLink{
+			ID: id,
+		}
+		if !ctx.isolatedCreation {
+			ctx.WorkItemLinks[i].LinkTypeID = ctx.WorkItemLinkTypes[0].ID
+			// this is the logic that ensures, each work item is only appearing
+			// in one link
+			ctx.WorkItemLinks[i].SourceID = ctx.WorkItems[2*i].ID
+			ctx.WorkItemLinks[i].TargetID = ctx.WorkItems[2*i+1].ID
+		}
+		ctx.runDeferredCreationFunctions(i, kindWorkItemLinks)
+		if ctx.isolatedCreation {
+			require.NotEqual(ctx.T, uuid.Nil, ctx.WorkItemLinks[i].LinkTypeID, "you must specify a work item link type for each work item link")
+			require.NotEqual(ctx.T, uuid.Nil, ctx.WorkItemLinks[i].SourceID, "you must specify a source work item for each work item link")
+			require.NotEqual(ctx.T, uuid.Nil, ctx.WorkItemLinks[i].TargetID, "you must specify a target work item for each work item link")
+		}
+		// default choice for creatorID: take it from the creator of the source work item
+		sourceWI, err := workitem.NewWorkItemRepository(ctx.db).LoadByID(ctx.ctx, ctx.WorkItemLinks[i].SourceID)
+		require.Nil(ctx.T, err, "failed to load the source work item in order to fetch a creator ID for the link")
+		creatorIDStr, ok := sourceWI.Fields[workitem.SystemCreator].(string)
+		require.True(ctx.T, ok, "failed to fetch the %s field from the source work item %s", workitem.SystemCreator, ctx.WorkItemLinks[i].SourceID)
+		creatorID, err := uuid.FromString(creatorIDStr)
+		require.Nil(ctx.T, err, "failed to convert the string \"%s\" to a uuid.UUID object", creatorIDStr)
+
+		err = link.NewWorkItemLinkRepository(ctx.db).CreateFromModel(ctx.ctx, ctx.WorkItemLinks[i], creatorID)
+		require.Nil(ctx.T, err, "failed to create work item link: %+v", ctx.WorkItemLinks[i])
+	}
+}
+
+func makeComments(ctx *TestContext) {
+	if ctx.info[kindComments] == nil {
+		return
+	}
+	ctx.Comments = make([]*comment.Comment, ctx.info[kindComments].numInstances)
+	for i := range ctx.Comments {
+		id := uuid.NewV4()
+		buf := `Lorem ipsum dolor sitamet, consectetur adipisicing elit, sed do eiusmod
```

Comment:
you could name this variable `loremIpsum` ;)

Reply:
Done